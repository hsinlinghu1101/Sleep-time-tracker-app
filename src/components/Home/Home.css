.sleep-record{
    border-style:solid;
    width:80%;
    border-color:#f2f7ff ;
}
.add{
    background-color: #10316b; 
}
.error{
    width:80%
}