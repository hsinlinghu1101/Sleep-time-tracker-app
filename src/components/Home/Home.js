import React from 'react'
import DataAPiService from '../../Services/data-api-service';
import NavBar from '../NavBar/NavBar'

export default class Home extends React.Component{

  static defaultProps={
    createDataSuccess:()=>{}
  }

  state={
    error:null
  }
  
  handleRecordSubmit=(event)=>{
    event.preventDefault();
    const{data_created, bed_time, wakeup_time }=event.target

    DataAPiService.postData({
      data_created: data_created.value,
      bed_time: bed_time.value, 
      wakeup_time: wakeup_time.value
    }) 

    .then(data =>{
      data_created.value=''
      bed_time.value=''
      wakeup_time.value=''
      this.props.createDataSuccess()
    })
    
    .catch(res =>{
      this.setState({
        error:JSON.stringify(res.error)
      })
    })
      
  }

  
    render(){
      const { error } = this.state
        return(
            <main role="main">
            <header>
              <NavBar/>
              <h1>Hi, {this.props.user_name} -Create New Record</h1>
              {error && <p>{error}</p>}
             
            </header>
            <section>
              <form id="sleep-record" onSubmit={this.handleRecordSubmit}>
                <div className="form-section">
                  <label htmlFor="getDate">Date</label>
                  <input type="date" name="data_created" required/>
               
                  <label htmlFor="bed-time">Go to Bed time: </label>
                  <input id="bed-time" type="time" name="bed_time" required/>
                  <label htmlFor="wakeup-time">Wake up time: </label>
                  <input id="wakeup-time" type="time" name="wakeup_time" required/>
                <button type="submit">Submit</button>
                </div>
              </form>
            </section>
          </main>
        )
    }
}