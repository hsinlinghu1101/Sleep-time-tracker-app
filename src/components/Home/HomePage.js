import React from 'react';
import Home from './Home';

export default class HomePage extends React.Component{
    static defaultProps={
        history:{
            push:()=>{}
        }
    }

    handleSubmitData=(data)=>{
        const { user_id }= this.props.match.params
        const {history} = this.props
        history.push(`/data/${user_id}`)
    }
    render(){
        return(
          <Home createDataSuccess={this.handleSubmitData}/>
      )
    }
        
    
}