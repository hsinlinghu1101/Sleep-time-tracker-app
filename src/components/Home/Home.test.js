import React from 'react';
import ReactDOM from 'react-dom';
import Home from './Home';
import {MemoryRouter as Router} from 'react-router-dom'
import renderer from 'react-test-renderer';

describe('Home', ()=>{
  it('renders without crashing', () => {
    
    const div = document.createElement('div');
   
    ReactDOM.render(<Router><Home /></Router>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
})
it('renders the UI as expected', () => {
  const tree = renderer
    .create(
        <Router><Home/></Router>    
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});