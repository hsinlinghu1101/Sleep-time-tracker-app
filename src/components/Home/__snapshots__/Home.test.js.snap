// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  role="main"
>
  <nav
    role="navigation"
  >
    <div
      className="nav"
    >
      Hi, 
      <div
        className="logout"
      >
        <a
          href="/"
          onClick={[Function]}
        >
           Logout 
        </a>
      </div>
    </div>
  </nav>
  <h2>
    Create New Record
  </h2>
  <form
    className="sleep-record"
    onSubmit={[Function]}
  >
    <div
      className="form-section"
    >
      <div>
        <label
          htmlFor="getDate"
        >
          Date - Go to Bed: 
        </label>
        <input
          id="data_created"
          name="data_created"
          required={true}
          type="date"
        />
        <br />
        <label
          htmlFor="bed-time"
        >
          Go to Bed time: 
        </label>
        <input
          id="bed-time"
          name="bed_time"
          required={true}
          type="time"
        />
      </div>
      <div>
        <label
          htmlFor="getDate"
        >
          Date - Wake Up: 
        </label>
        <input
          id="data_wakeup"
          name="data_wakeup"
          required={true}
          type="date"
        />
        <br />
        <label
          htmlFor="wakeup-time"
        >
          Wake up time: 
        </label>
        <input
          id="wakeup-time"
          name="wakeup_time"
          required={true}
          type="time"
        />
      </div>
      <button
        className="btn add"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</main>
`;
